### PR TITLE
Fix Product Rating Block not showing when manually inserted to Single Product Block

### LINKED ISSUE #9333: Product Rating Block: When it is added manually inside the Single Product Block isn't visible on the frontend
## Describe the bug

When the Product Rating block is added manually inside the Single Product Block, the preview isn't correct nor visible on the front end.

## To reproduce

1. Create a new post.
2. Add the Single Product Block and select a product with a review.
3. Be sure that the preview is correct.
4. Remove the Product Rating block.
5. Re-add the Product Rating block.
6. You can see that the preview isn't correct.
7. Save the post.
8. Visit the post.
9. The Product Rating isn't visible.

## Expected behavior

When the Product Rating is added, the preview should be correct, and the block should be visible on the frontend.

## Screenshots

https://user-images.githubusercontent.com/4463174/235895138-4c55f07c-d006-4980-9e6c-45d09208bd75.mp4

### PR DESCRIPTION
**Pull request by thealexandrelara**
### Summary
This solves an issue that was being caused when manually adding the Product Rating block to the Single Product block. The "isDescendentOfSingleProductBlock" attribute is what defines how the Product Rating block will behave depending on the context in which it is inserted. The issue is that when the Product Rating block was manually added, this attribute remained with a `false` value, so this way the block didn't know that it was inside the context of the Single Product block. 

### <samp>🤖 Generated by Copilot at 8b7b69e</samp>

* Import `useSelect` hook from `@wordpress/data` package to select data from WordPress data store ([link](https://github.com/woocommerce/woocommerce-blocks/pull/9413/files?diff=unified&w=0#diff-4e154173c83a90857629f0bdf60e0af8dbd33560750747056a1535ad9132e966R13))
* Modify `Edit` component to use `useSelect` hook to check if block is a descendent of `woocommerce/single-product` block and store result in new attribute `isDescendentOfSingleProductBlock` ([link](https://github.com/woocommerce/woocommerce-blocks/pull/9413/files?diff=unified&w=0#diff-4e154173c83a90857629f0bdf60e0af8dbd33560750747056a1535ad9132e966L36-R55))
* Conditionally render rating element based on `isDescendentOfSingleProductBlock` and `isDescendentOfQueryLoop` attributes ([link](https://github.com/woocommerce/woocommerce-blocks/pull/9413/files?diff=unified&w=0#diff-4e154173c83a90857629f0bdf60e0af8dbd33560750747056a1535ad9132e966L36-R55))
* Update `useEffect` hook to set both attributes when component mounts or updates ([link](https://github.com/woocommerce/woocommerce-blocks/pull/9413/files?diff=unified&w=0#diff-4e154173c83a90857629f0bdf60e0af8dbd33560750747056a1535ad9132e966L36-R55))
* Extend `BlockAttributes` interface to include new attribute ([link](https://github.com/woocommerce/woocommerce-blocks/pull/9413/files?diff=unified&w=0#diff-ce0518f348a8ec5bfbdf899d766dffba0e0cb297d927980ea53141845e9e5399R4))

Fixes #9333 

#### Accessibility

- [ ] I've tested using only a keyboard (no mouse)
- [ ] I've tested using a screen reader
- [ ] All animations respect [`prefers-reduced-motion`](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-reduced-motion)
- [ ] All text has [at least a 4.5 color contrast with its background](https://webaim.org/resources/contrastchecker/)

#### Other Checks

- [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] This PR adds/removes an experimental interfaces and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
- [ ] I tagged two reviewers because this PR makes queries to the database or I think it might have some security impact.

#### Automated Tests
* [ ] Changes in this PR are covered by Automated Tests.
  * [ ] Unit tests
  * [ ] E2E tests

#### User Facing Testing

1. Log in to your WordPress dashboard.
2. On the left-hand side menu, click on 'Posts' and then 'Add New' to create a new post. Alternatively, if you want to add the block to an existing post, you can simply find that post and click 'Edit'.
3. Inside the post editor, click on the '+' button, usually found at the top left of the editing space or within the content area itself, to add a new block.
4. In the block library that pops up, you can search for the 'Single Product' block. You can do this by typing 'Single Product' into the search bar at the top of the block library.
5. Click on the 'Single Product' block to add it to your post. You'll be prompted to select a product from your WooCommerce store to feature in the block.
6. Once you've selected a product, the block will be inserted into your post and will display information about the product you've selected.
7. Click on the 'Single Product' block in your post to select it.
8. In the blocks list on the left-hand side, you should see different components of the 'Single Product' block that you can select, remove, or modify. These components typically include the product image, title, price, rating, summary, and button.
9. Find the 'Product Rating' component, click on the three dots to the right side, click Remove. This will remove the product rating from the block.
10. To add the Product Rating back into the Single Product block, simply click on the 'Single Product' block to select it again.
11. With the block selected, add the Product Rating block inside it.

* [x] Do not include in the Testing Notes 

### WooCommerce Visibility

* [ ] WooCommerce Core
* [ ] Feature plugin
* [x] Experimental

### CODE REVIEW

File: assets/js/atomic/blocks/product-elements/rating/edit.tsx [hunk 1]
```diff
@@ -33,11 +34,25 @@ const Edit = ( {
 		...context,
 	};
 	const isDescendentOfQueryLoop = Number.isFinite( context.queryId );
+	const { isDescendentOfSingleProductBlock } = useSelect( ( select ) => {
```
Comment (nefeline) [inline assets/js/atomic/blocks/product-elements/rating/edit.tsx:37]: Would it be worth having this logic stored as a React hook for shared usage across all inner blocks for the Single Product? Could other blocks, such as price and add to cart, also benefit from this change?

**Review by nefeline — COMMENTED** [review]
Works like a charm, well done 🎉 ! I've just left a question.

File: assets/js/atomic/blocks/product-elements/rating/edit.tsx (hunk 1 above)
Comment (thealexandrelara) [reply assets/js/atomic/blocks/product-elements/rating/edit.tsx:37]: Yes! I agree with this, today I opened another PR to an issue that I was investigating and I found out that it was being caused by the same thing, so I created a hook [in that PR](https://github.com/woocommerce/woocommerce-blocks/pull/9446) that I'll be using to refactor this and other blocks that need it

Comment (nefeline) [reply assets/js/atomic/blocks/product-elements/rating/edit.tsx:37]: Cool! Looking forward to having https://github.com/woocommerce/woocommerce-blocks/pull/9446 merged so this PR can benefit from this update.

**Review by nefeline — APPROVED** [review]
:shipit:

**Pull request merged**